pkg/plugin/datasource.go

```diff
@@ -43,8 +45,23 @@ func NewDatasource(settings backend.DataSourceInstanceSettings) (instancemgmt.In
 		return nil, fmt.Errorf("failed to initizlize Spice AI client: %w", err)
 	}
 
+	opts, err := settings.HTTPClientOptions()
+	if err != nil {
+		return nil, fmt.Errorf("http client options: %w", err)
+	}
+	opts.Headers = map[string]string{
+		"Content-Type": "application/json",
+		"X-API-Key":    apiKey,
```

Comment:
Add `Accept-Encoding: gzip, deflate`

Reply:
Done, also updated response parser to support new encodings